mypy/lookup.py

```diff
@@ -0,0 +1,59 @@
+"""
+This is a module for various lookup functions:
+functions that will find a semantic node by its name.
+"""
+
+from mypy.nodes import MypyFile, SymbolTableNode, TypeInfo
+from typing import Dict, Optional
+
+# TODO: gradually move existing lookup functions to this module.
+
+
+def lookup_fully_qualified(name: str, modules: Dict[str, MypyFile],
```

Comment:
Can you verify that this new lookup method doesn't cause a significant slowdown using compiled mypy? The original method was simpler and if there is a performance impact, it may make sense to keep the original method as an alternative/optimization (but carefully document it as such).

Reply:
I decided to keep the old look-up function in `semanal.py` anyway. It doesn't really belong to this PR. We can make a separate PR that cleans up all lookup related things.